docs/Easyconfigs_index.rst

```diff
@@ -0,0 +1,151 @@
+and will contain a list of (relative) paths of all files in that directory,
+along with some metadata: the time at which the index file has created,
+and a timestamp that indicates how long the index is considered to be valid (see :ref:`easyconfigs_index_max_age`).
+
+.. note::
+  Make sure to create an index for the correct path.
+
+  When searching for easyconfig files, EasyBuild will *not* recurse into subdirectories of the locations
+  it considers (see :ref:`robot_search_path`), other than the ones it considers based on the name of the
+  software for which it is trying to find an easyconfig file (like ``t/TensorFlow/`` when searching for an
+  easyconfig file for ``TensorFlow``).
+
+  Hence, if the directory where your easyconfig files are housed in has an ``easybuild/easyconfigs`` subdirectory
+  (for example, a working copy of the ``easybuild-easyconfigs`` repository), create the index *in* that
+  subdirectory, rather than in the higher-level directory.
+
+
+.. _easyconfigs_index_create_example:
+
+Example of creating an index
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code::
+
+  $ eb --create-index $HOME/easybuild-easyconfigs/easybuild/easyconfigs
+  == temporary log file in case of crash /tmp/eb-tUu6_w/easybuild-SKBnVO.log
+  Creating index for /home/example/easybuild-easyconfigs/easybuild/easyconfigs...
+  == found valid index for /home/example/easybuild-easyconfigs/easybuild/easyconfigs, so using it...
+  Index created at /home/example/easybuild-easyconfigs/easybuild/easyconfigs/.eb-path-index (738 files)
+
+  $ head -n 5 $HOME/easybuild-easyconfigs/easybuild/easyconfigs/.eb-path-index
+  # created at: 2020-04-13 14:19:57.008981
+  # valid until: 2020-04-20 14:19:57.008981
+  a/Arrow/Arrow-0.16.0-intel-2019b-Python-3.7.4.eb
+  b/Boost/Boost-1.71.0-gompi-2019b.eb
+  b/bokeh/bokeh-1.4.0-foss-2019b-Python-3.7.4.eb
+
+
+.. note::
+  The "``found valid index ...``" message being printed occurs because EasyBuild tries to load the index file
+  right after creating it, as a sanity check.
+
+
+.. _easyconfigs_index_update:
+
+Updating an existing index (``--create-index --force``)
+-------------------------------------------------------
+
+To update an existing index, you can use ``--create-index --force``.
+
+Without using ``--force``, EasyBuild will refuse to overwrite the existing index file::
+
+  $ eb --create-index $HOME/easybuild-easyconfigs/easybuild/easyconfigs
+  == temporary log file in case of crash /tmp/eb-tUu6_w/easybuild-SKBnVO.log
+  Creating index for /home/example/easybuild-easyconfigs/easybuild/easyconfigs...
+  ERROR: File exists, not overwriting it without --force: /home/example/easybuild-easyconfigs/easybuild/easyconfigs/.eb-path-index
+
+
+.. _easyconfigs_index_use:
+
+Using index files
+-----------------
+
+EasyBuild will automatically pick up & use an index file if it runs into it when searching for
+easyconfig files or patches, rather than walking the directory to check whether the file it's looking for
```

Comment:
```suggestion
easyconfig files or patches. Once an index file is found, it will preferred over walking through the directory tree to find the target easyconfig,
```